pkg/services/hcloud/server/server.go

```diff
@@ -392,6 +392,9 @@ func (s *Service) createServer(ctx context.Context) (*hcloud.Server, error) {
 	server, err := s.scope.HCloudClient.CreateServer(ctx, opts)
 	if err != nil {
 		hcloudutil.HandleRateLimitExceeded(s.scope.HCloudMachine, err, "CreateServer")
+		if hcloud.IsError(err, hcloud.ErrorCodeResourceLimitExceeded) {
+			conditions.MarkFalse(s.scope.HCloudMachine, infrav1.ServerCreateSucceededCondition, infrav1.RateLimitExceededReason, clusterv1.ConditionSeverityError, err.Error())
```

Comment:
the reason  RateLimitExceeded is wrong. You should create a new one "ServerLimitExceededReason"